### PR TITLE
docs: create Contributing Guide

### PR DESCRIPTION
**Pull request by machikoyasuda**
Since there were 2 very similar pages (Contributing and Pull request reviews), I combined them and renamed them as the `Contributing Guide`, and styled it as a tutorial with steps.

screen shot 2018-01-11 at 5 37 38 pm

https://docs.reactioncommerce.com/reaction-docs/machikoyasuda-merging-contributing-guide/contributing-to-reaction

### CODE REVIEW

File: developer/contributing.md [hunk 1]
```diff
@@ -32,3 +30,45 @@ Let us know how we can make the contribution process easier. We want to find the
 -   Should passing a linter code review and follow Reaction style guidelines.
 -   Contributors should review the CLA.
 -   Code reviewed before merge acceptance.
+
+### Here's what to expect when you make a pull request to Reaction
+
+As soon as pull requests are pushed, [BitHound](https://www.bithound.io/github/reactioncommerce/reaction), [CircleCI](https://circleci.com/gh/reactioncommerce/reaction) and [Synk](https://snyk.io/) run automated tests. All checks should pass successfully to ensure:
```
Comment (zenweasel) [inline developer/contributing.md]: This part about Bithound is no longer true. And it's CircleCI that run automated tests not Bithound. Maybe we shouldn't name the service but just say that the code will be checked for lint error?

File: developer/contributing.md [hunk 2]
```diff
@@ -32,3 +30,45 @@ Let us know how we can make the contribution process easier. We want to find the
...
+-   Dependencies are not failing.
+-   All tests pass.
+-   No new security vulnerabilities are introduced.
+
+You can run these checks yourself by running these commands:
+
+-   `eslint .` for linting
+-   `reaction test` for all tests
+
+### Core team pull request review
```
Comment (zenweasel) [inline developer/contributing.md]: The whole "core team review" is no longer true. Probably best to just say that the Community team will review all community PRs as soon as possible. All of these notes apply equally to both Core and Community PR's.

**Comment by machikoyasuda** [comment]
@zenweasel How do these changes look? 

Removed explicit reference to BitHound and Core team:

https://github.com/reactioncommerce/reaction-docs/pull/435/commits/58867deabb5a02b72677cd33a930a2c52f3dd13b

**Comment by machikoyasuda** [comment]
I can manually merge this once it's good to go @zenweasel ! `redoc.json` has caused major conflicts :(

**Pull request merged**